pkg/configurations/configurations.go

```diff
@@ -0,0 +1,159 @@
+package configurations
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/gobwas/glob"
+
+	"github.com/fsnotify/fsnotify"
+
+	"github.com/thoas/go-funk"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	"github.com/spf13/viper"
+)
+
+const defaultInterfacesFilter = "veth*"
+const defaultNodeNetworkStateRefresh = 5
+
+var (
+	log   = logf.Log.WithName("configurations")
+	mutex = sync.Mutex{}
+)
+
+// Struct to match config file
+type config struct {
+	NodeNetworkRefreshInterval int    `mapstructure:"node_network_state_refresh_interval"`
+	InterfaceFilter            string `mapstructure:"interfaces_filter"`
+	InterfacesFilterGlob       glob.Glob
+}
+
+var c = config{
+	NodeNetworkRefreshInterval: defaultNodeNetworkStateRefresh,
+	InterfaceFilter:            defaultInterfacesFilter,
+}
+
+type configWatcher struct {
+	configPath string
+	v          *viper.Viper
+}
+
+func NewConfigWatcher() *configWatcher {
+	configPathTemp, isSet := os.LookupEnv("CONFIG_PATH")
+	if !isSet {
+		panic("CONFIG_PATH is mandatory")
+	}
+	cw := configWatcher{
+		configPath: configPathTemp,
+	}
+	cw.v = viper.New()
+	return &cw
+}
+
+func GetCurrentConfig() config {
+	return c
```

Comment:
We need the mutex for reading.

Reply:
Why we need mutex for reading? Reading is not atomic?

Reply:
Not while somebody is writting to it. Afaik go has no guarantees on atomicity of string reading/writting. If you find a trustworthy source saying otherwise, please let me know.